test/end-to-end/Drawer-Login.test.ts

```diff
@@ -0,0 +1,190 @@
+import { Selector } from 'testcafe'
+import {
+  otp,
+  rootLocation,
+  shortUrl,
+  subUrl,
+  testEmail,
+  transferEmail,
+} from './util/config'
+import {
+  activeSwitch,
+  clickAway,
+  closeButtonSnackBar,
+  createLinkButton,
+  drawer,
+  generateUrlImage,
+  helperText,
+  inactiveWord,
+  linkErrorSnackBar,
+  linkTransferField,
+  loginButton,
+  longUrl,
+  longUrlTextField,
+  shortUrlTextField,
+  signInButton,
+  signOutButton,
+  successSnackBar,
+  transferButton,
+  urlSaveButton,
+  urlUpdatedToaster,
+  userModal,
+  userModalCloseButton,
+} from './util/helpers'
+import LoginProcedure from './util/Login-Procedure'
+
+// eslint-disable-next-line no-undef
+fixture(`Drawer Page`)
+  .page(`${rootLocation}`)
+  .beforeEach(async (t) => {
+    await LoginProcedure(t)
+  })
+
+test('Drawer functionality test.', async (t) => {
+  await t.click(createLinkButton.nth(0)).click(generateUrlImage)
+
+  const generatedUrl = await shortUrlTextField.value
+  const linkRow = Selector(`h6[title="${generatedUrl}"]`)
+
+  await t.typeText(longUrlTextField, `${shortUrl}`)
+
+  if (await createLinkButton.nth(2).exists) {
+    await t.click(createLinkButton.nth(2))
+  } else {
+    await t.click(createLinkButton.nth(1))
+  }
+
+  await t
+    .click(linkRow)
+    .expect(drawer.exists)
+    .ok()
+    // Drawer should open with the correct long url and state when a short url row is clicked
+    .expect(longUrl.value)
+    .eql(`${shortUrl}`)
+
+  await t
+    .click(drawer.child(2).child('main').child('button'))
+    // Drawer can be closed on clickaway or when the close button is clicked.
+    .expect(drawer.exists)
+    .notOk()
```

Comment:
From the documentation, the equality (notOk) takes the previous assertion value and matches it. In this case, click is not a type of assertion and more of a selection. 
https://devexpress.github.io/testcafe/documentation/guides/basic-guides/assert.html

If click fails, it will move on to the assertion (expect drawer exists) - which will fail because the drawer is still open.